ert_shared/plugins/workflow_config.py

```diff
@@ -0,0 +1,96 @@
+import inspect
+import os
+import tempfile
+
+
+class WorkflowConfigs(object):
+    """
+    Top level workflow config object, holds all workflow configs.
+    """
+
+    def __init__(self):
+        self._temp_dir = tempfile.mkdtemp()
+        self._workflows = []
+
+    def add_workflow(self, ert_script, name=None):
+        """
+
+        :param ert_script: class which inherits from ErtScript
+        :param name: Optional name for workflow (default is name of class)
+        :return: Instantiated workflow config.
+        :type: :class:`ert_shared.plugins.workflow_config.WorkflowConfig`
+        """
+        workflow = WorkflowConfig(ert_script, self._temp_dir, name)
+        self._workflows.append(workflow)
+        return workflow
+
+    def get_workflows(self):
+        configs = {}
+        for workflow in self._workflows:
+            configs[workflow.name] = workflow.config_path
+        return configs
+
+
+class WorkflowConfig(object):
+    """
+    Single workflow configuration object
+
+    """
+
+    def __init__(self, ertscript_class, tmpdir, name=None):
+        """
+        :param ertscript_class: Class inheriting from ErtScript
+        :param tmpdir: Where workflow config is generated
+        :param name: Optional name for workflow, default is class name
+        """
+        self.func = ertscript_class
+        self.name = self._get_func_name(ertscript_class, name)
+        self.function_dir = os.path.abspath(inspect.getfile(ertscript_class))
+        self.config_path = self._write_workflow_config(tmpdir)
+        self._description = ertscript_class.__doc__ if ertscript_class.__doc__ else ""
+        self._examples = None
+        self._parser = None
+
+    @property
+    def description(self):
```

Comment:
I changed the settable attributes to `@property` so that we can more easily keep documentation up to date. As this is an important part of the API, I figured that would be a good way to go about it, but my mind can be changed.